src/seedu/addressbook/AddressBook.java

```diff
@@ -362,14 +367,28 @@ public static String executeCommand(String userInputString) {
             return executeClearAddressBook();
         case COMMAND_HELP_WORD:
             return getUsageInfoForAllCommands();
+        case COMMAND_SORT_DATA:
+        	return sortAddressBook();
+        	
         case COMMAND_EXIT_WORD:
             executeExitProgramRequest();
         default:
             return getMessageForInvalidCommandInput(commandType, getUsageInfoForAllCommands());
         }
     }
-
-    /**
+    //sorting method
+    private static String sortAddressBook() {
+    	Collections.sort(allPersons.subList(0, allPersons.size()), new Comparator<String[]>() {
+    		public int compare(String[] person2, String[] person1) {
+    			return person2[PERSON_DATA_INDEX_NAME].compareToIgnoreCase(person1[PERSON_DATA_INDEX_NAME]);
+    		}
+    	});
+    	savePersonsToFile(allPersons, storageFilePath);
+    	executeListAllPersonsInAddressBook();
+		return MESSAGE_SORT_SUCCESSFUL;
+	}
```

Comment:
indentation not according to coding style. use autoformat after every change. Add test cases for sort.